src/compiler/program.ts

```diff
@@ -104,14 +104,14 @@ namespace ts {
         };
     }
 
-    export function getPreEmitDiagnostics(program: Program, sourceFile?: SourceFile): Diagnostic[] {
-        let diagnostics = program.getOptionsDiagnostics().concat(

```

Comment:
why are we passing a cancellation token here. the program has access to the host, and the host has a cancellation token already.